src/cargo/util/toml/mod.rs

```diff
@@ -987,18 +987,39 @@ pub fn to_real_manifest(
         .transpose()?
         .unwrap_or_default();
     let empty_features = BTreeMap::new();
+    let deps_map = deps
+        .iter()
+        .filter(|dep| dep.is_transitive() || dep.specified_req())
+        .map(|dep| (dep.name_in_toml(), dep))
+        .collect::<BTreeMap<InternedString, &Dependency>>();
```

Comment:
A red flag for this commit is that no tests changed.

`to_real_manifest` doesn't strip any dependencies so features don't need to be stripped.

Reply:
Thanks @epage , i'll then drop this commit later.